### PR TITLE
Update descriptions for VS Code extensions

### PR DESCRIPTION
**Pull request by ruthemmanuelle**
### What does this PR do?
Updates the extensions' descriptions to match Salesforce product names.

### What issues does this PR fix or reference?
@W-4198892@

### CODE REVIEW

File: packages/salesforcedx-vscode-core/package.json [hunk 1]
```diff
@@ -1,8 +1,8 @@
 {
   "preview": true,
   "name": "salesforcedx-vscode-core",
-  "displayName": "SFDX CLI integration for Visual Studio Code",
-  "description": "Provides integration with the SFDX CLI",
+  "displayName": “Salesforce CLI integration for Visual Studio Code",
```
Comment (vazexqi) [inline packages/salesforcedx-vscode-core/package.json]: @ruthemmanuelle You used the wrong " here.

Comment (ruthemmanuelle) [inline packages/salesforcedx-vscode-core/package.json]: No wonder there's weird highlighting. Thanks. I'll fix it.

**Comment by ruthemmanuelle** [comment]
@vazexqi - I've fixed the quotation mark.

**Comment by vazexqi** [comment]
LGTM. Please squash and merge once the checks complete (I don't think they should fail since there are no code changes)

**Pull request merged**